src/encoded/types/workflow.py

```diff
@@ -808,19 +802,28 @@ def map_run_data_to_io_arg(step_io_arg, wfr_runtime_inputs, io_type):
 
             if len(matched_runtime_io_data) > 0:
                 matched_runtime_io_data = sorted(matched_runtime_io_data, key=lambda io_object: io_object.get('ordinal', 1))
+
+                # List of file or Item UUIDs.
+                value_list = []
+
+                # Aligned-to-file-list-indices list of file metadata(s)
+                meta_list = []
+
+                for io_dict in matched_runtime_io_data:
+                    # Quality Metrics might be saved with `value_qc` instead of `value`
+                    linked_to_item_uuid = io_dict.get('value') or io_dict.get('value_qc')
+                    value_list.append(linked_to_item_uuid)
+                    # Add all remaining properties from dict in (e.g.) 'input_files','output_files',etc. list.
+                    # Contains things like dimension, ordinal, file_format, and so forth.
+                    meta_dict = { k:v for (k,v) in io_dict.items() if k not in [ 'value', 'value_qc', 'type', 'workflow_argument_name' ] }
+                    # There is a chance we do not have the file_format in the input_files list. Most often this would occur if multiple
+                    # files in input argument list 
+                    meta_list.append(meta_dict)
+
                 step_io_arg['run_data'] = {
-                    value_field_name : [
-                        # List of file UUIDs
-                        # Also, Quality Metrics might be saved with `value_qc` instead of `value`
-                        (io_object.get('value') or io_object.get('value_qc')) for io_object in matched_runtime_io_data
-                    ],
+                    value_field_name : value_list,
                     "type" : io_type,
-                    "meta" : [ # Aligned-to-file-list list of file metadata
-                        {   # All remaining properties from dict in (e.g.) 'input_files','output_files',etc. list.
-                            k:v for (k,v) in p.items()
-                            if k not in [ 'value', 'value_qc', 'type', 'workflow_argument_name' ]
-                        } for p in matched_runtime_io_data
-                    ]
+                    "meta" : meta_list
                 }
```

Comment:
No logic has changed in the above lines, just refactored to make more clear that indices of run_data.meta and run_data.file (or run_data.value, if is parameter and not linkto) are aligned.